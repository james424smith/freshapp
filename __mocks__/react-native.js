import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
