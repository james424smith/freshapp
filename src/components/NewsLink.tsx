import React from "react";
import { View, TouchableOpacity } from "react-native";
import Text from "./StyledText";
import styles from "../styles/AuthStyles";
import { NEWS_ROUTE } from "../constants/routes";

type Props = { navigate: (arg: string) => void };

const NewsLink = (props: Props) => {
  const { navigate } = props;

  return (
    <View>
      <View style={styles.newsLinkView}>
        <TouchableOpacity
          testID={"news-button"}
          onPress={() => navigate(NEWS_ROUTE)}
        >
          <Text style={styles.newsLink}>News</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

export default NewsLink;
