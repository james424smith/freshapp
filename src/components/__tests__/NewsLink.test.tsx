import React from "react";
import NewsLink from "../NewsLink";
import { render, cleanup, fireEvent } from "@testing-library/react-native";

describe("NewsLink snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });

  it("should render correct ui and click the news-button", () => {
    const navigate = jest.fn();
    const { toJSON, getByTestId } = render(<NewsLink navigate={navigate} />);
    fireEvent.press(getByTestId("news-button"));
    expect(navigate).toHaveBeenCalledWith("News");
    expect(toJSON()).toMatchSnapshot();
  });
});
