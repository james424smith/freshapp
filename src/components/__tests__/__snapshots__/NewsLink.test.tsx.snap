// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsLink snapshot test should render correct ui and click the news-button 1`] = `
<View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="news-button"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Object {
              "color": "#78849E",
              "textDecorationLine": "underline",
            },
          ]
        }
      >
        News
      </Text>
    </View>
  </View>
</View>
`;
